packages/model-viewer/src/features/environment.ts

```diff
@@ -126,6 +126,27 @@ export const EnvironmentMixin = <T extends Constructor<ModelViewerElementBase>>(
         this[$cancelEnvironmentUpdate] = null;
       }
 
+      // Update background color before to get texture utils because renderer singleton depends on background transparency
+      if (skyboxImage == null)
+      {
+        if (backgroundColor === 'transparent') {
+          if (this[$scene].background != null) {
+            this.disconnectedCallback();
+            delete this[$scene].background;
+            this.connectedCallback();  
+          }
+        }
+        else {
+          const mustUpdateRenderer = this[$scene].background == null;
+          mustUpdateRenderer && this.disconnectedCallback();
```

Comment:
If these lifecycle methods are being invoked in order to get the renderer to update, then we could use `this[$needsRender]()` to provoke a re-render and skip calling the lifecycle methods.

Reply:
The objective is not really to update the rendering but to unsubscribe listeners and unregister the scene from the old renderer before switching to the new renderer. So the accessor $renderer must return the old renderer to unsubscribe the listener, then the same accessor must return the new renderer to subscribe the listener. This is why there is the test to null on the background performed in the accessor.